cli/src/program.rs

```diff
@@ -337,8 +343,8 @@ impl ProgramSubCommands for App<'_, '_> {
                         ),
                 )
                 .subcommand(
-                    SubCommand::with_name("dump")
-                        .about("Write the program data to a file")
+                    SubCommand::with_name("dump-executable")
+                        .about("Write the executable program buffer data to a file")
```

Comment:
Can you add `.alias("dump")` or keep `"dump"` around with a deprecation notice? Also, can you update the docs at docs/src/cli/deploy-a-program.md to specify using `dump-executable` instead?